SimCalorimetry/HcalTrigPrimAlgos/src/HcalTriggerPrimitiveAlgo.cc

```diff
@@ -580,12 +582,22 @@ void HcalTriggerPrimitiveAlgo::analyzeQIE11(IntegerCaloSamples& samples,
         output[ibin] = 0;
       }
     } else {
-      output[ibin] = std::min<unsigned int>(sum[idx], QIE11_MAX_LINEARIZATION_ET);
+      bool isPeak = (sum[idx] > sum[idx - 1] && sum[idx] >= sum[idx + 1]);
+      if (force_saturation[idx - 1] || force_saturation[idx] || force_saturation[idx + 1])
+        isPeak = force_saturation[idx] && !force_saturation[idx - 1];
```

Comment:
Silly question, not sure I fully I understand what's happening, but should there be some sort of `force_saturation` check for `idx+1`?

Reply:
Hi @aloeliger ,
Thank you for your review. You are correct, the saturation condition should be checked for the BX-1 _and_ BX+1 time samples. I will make this change, and also add some commenting in this region of the code to clarify.